synapse/module_api/__init__.py

```diff
@@ -1452,6 +1452,65 @@ async def get_monthly_active_users_by_service(
             start_timestamp, end_timestamp
         )
 
+    async def create_room(
+        self,
+        user_id: str,
+        config: JsonDict,
+        ratelimit: bool = True,
+        creator_join_profile: Optional[JsonDict] = None,
+    ) -> Tuple[dict, int]:
+        """Creates a new room.
+
+        Args:
+            user_id:
+                The user who requested the room creation.
+            config : A dict of configuration options. See "Request body" of:
+                https://spec.matrix.org/latest/client-server-api/#post_matrixclientv3createroom
+            ratelimit: set to False to disable the rate limiter.
+
+            creator_join_profile:
+                Set to override the displayname and avatar for the creating
+                user in this room. If unset, displayname and avatar will be
+                derived from the user's profile. If set, should contain the
+                values to go in the body of the 'join' event (typically
+                `avatar_url` and/or `displayname`.
+
+        Returns:
+                First, a dict containing the keys `room_id` and, if an alias
+                was, requested, `room_alias`. Secondly, the stream_id of the
+                last persisted event.
```

Comment:
```suggestion
                A tuple. The first element is a dict containing the keys `room_id`
                and, if an alias was, requested, `room_alias`. The second element
                is the stream_id of the last persisted event.
```

Reply:
I'm also unsure it makes a lot of sense to give stream IDs to modules. Streams should be internal to Synapse and I'm not sure why a module should be manipulating them.

Reply:
Ah, fair point. Changed here:

- 549b685
- cdf27d5